File: 1.0/src/cz/mg/vulkan/PFNvkDestroyPipelineCache.java
package cz.mg.vulkan;

public class PFNvkDestroyPipelineCache extends VkFunctionPointer {
    public PFNvkDestroyPipelineCache() {
    }

    public PFNvkDestroyPipelineCache(VkMemory vkmemory) {
        super(vkmemory);
    }

    public PFNvkDestroyPipelineCache(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkDestroyPipelineCache(long value) {
        setValue(value);
    }

    public PFNvkDestroyPipelineCache(VkInstance instance) {
        super(instance, new VkString("vkDestroyPipelineCache"));
    }

    public void call(VkDevice device, VkPipelineCache pipelineCache, VkAllocationCallbacks pAllocator){
        call(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddress(), pipelineCache != null ? pipelineCache.getVkAddress() : VkPointer.getNullAddress(), pAllocator != null ? pAllocator.getVkAddress() : VkPointer.NULL);
    }

    private static native void call(long vkaddress, long device, long pipelineCache, long pAllocator);
}

File: 1.0/src/c/cz/mg/vulkan/PFN_vkDestroyPipelineCache.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkDestroyPipelineCache_callNative(JNIEnv* env, jclass jc, jlong address, jlong device, jlong pipelineCache, jlong pAllocator){
    (void)env;
    (void)jc;
    PFN_vkDestroyPipelineCache f = (PFN_vkDestroyPipelineCache)jniLongToFunctionPointer(address);
    f(
        *((VkDevice*)jniLongToPointer(device)),
        *((VkPipelineCache*)jniLongToPointer(pipelineCache)),
        ((VkAllocationCallbacks*)jniLongToPointer(pAllocator))
    );
}

